### ISSUE TITLE
Category 2 of Wallet.calculateFee() is never used

### ISSUE THREAD

**Issue by oscarguindzberg** [bitcoinj/bitcoinj#1346]
Category 2 is
"By adding change, there is an increase in fee (change < 0.01 COINS)"

But that category is never used in practice.

I am not saying category 2 should be removed, we just need to use it.

If you run WalletTest.java with a breakpoint here https://github.com/bitcoinj/bitcoinj/blob/1fd8b90f428b701421e1c0984ad16b5fbd129cc8/core/src/main/java/org/bitcoinj/wallet/Wallet.java#L4876
Eclipse never stops at the breakpoint.

**Comment by oscarguindzberg** [comment]
Maybe related to #1345

**Comment by oscarguindzberg** [comment]
After
https://github.com/bitcoinj/bitcoinj/blob/1fd8b90f428b701421e1c0984ad16b5fbd129cc8/core/src/main/java/org/bitcoinj/wallet/Wallet.java#L4839-L4840

either "needAtLeastReferenceFee" (value copied form req.ensureMinRequiredFee) is false or "fees" is set to at least Transaction.REFERENCE_DEFAULT_MIN_TX_FEE

Then, this code will never evaluate to true
https://github.com/bitcoinj/bitcoinj/blob/1fd8b90f428b701421e1c0984ad16b5fbd129cc8/core/src/main/java/org/bitcoinj/wallet/Wallet.java#L4873-L4874

**Comment by schildbach** [comment]
It's very well possible I broke something when I did the various fee-related refactorings a year ago. I wonder if we should port that code from scratch from Bitcoin Core again? I am badly missing a "recipient pays fee" flag which Core has since a while already (we have the emptyWallet flag but that covers only a tiny edge case).

Anyway, if you're confident this code is unused feel free to either fix the cause or remove the code. The current fee calculation is in production since a year without serious problems. I'm happy for any chance that disentangles this code a bit. I had a hard time a year ago…
